inc/zoo/swar/SWAR.h

```diff
@@ -12,6 +12,14 @@
 
 namespace zoo { namespace swar {
 
+template <int NBits, typename T>
+struct SWAR;
+
+template <int NumBits, typename BaseType> struct Literals_t {};
+
+template <int NumBits, typename BaseType>
+constexpr Literals_t<NumBits, BaseType> Literals{};
```

Comment:
I dislike that we've got to do a packaged sub-type to...get a type match? I haven't tried to avoid that before, but it seems like something probably tractable.